tests/migration/storage.go

```diff
@@ -0,0 +1,152 @@
+package migration
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/stretchr/testify/require"
+)
+
+// Options groups optional parameters for TestRemoteStorageMigration.
+type Options struct {
+	// Path to the directory containing source code of the NeoFS NNS contract
+	// overriding the default one in step 1.
+	NNSSourceCodeDir string
+
+	// OnNNSDeploy allows to modify DeployConfig dynamically after NeoFS NNS
+	// contract deployment.
+	OnNNSDeploy func(*DeployConfig, util.Uint160)
+
+	// Group of NeoFS contracts on which the tested one depends. If set, all
+	// superior contracts are pre-handled locally before step 3 in order.
+	SuperiorContracts []SuperiorContractConfig
+
+	// Path to the directory containing source code of the tested NeoFS contract
+	// overriding the default one in step 4.
+	SourceCodeDir string
+
+	// Configuration of deployment of the tested contract for step 4.
+	DeployConfig DeployConfig
+
+	// Update failure scenarios reproduced when trying to update before 5. All
+	// scenarios are tested in order, so they should follow check order of the
+	// contract update.
+	UpdateFailures []UpdateFailScenario
+
+	// Configuration with which the tested contract should be deployed without
+	// problems in step 5. ValidUpdateConfig is also passed to any deploy callback
+	// of SuperiorContracts.
+	ValidUpdateConfig UpdateConfig
+
+	// Update data transition model replacing expected storage emptiness in step 6.
+	// If set, post-update contract storage items must be compliant with the model.
+	// Note that all expected value must be reflected in the model, otherwise test
+	// fails.
+	UpdateModel UpdateModel
+}
+
+// TestRemoteStorageMigration checks that the contract data relevant for the
+// remote NeoFS blockchain instance migrates as expected. On success,
+// TestRemoteStorageMigration just returns. All problems are reported via
+// provided testing.TB. The data is pulled using provided endpoint of the Neo
+// RPC server. The tested contract is identified by its name.
+//
+// Remote blockchain prerequisites:
+//   - NeoFS NNS contract is presented with ID = 1
+//   - tested contract is deployed and registered in the NeoFS NNS contract with
+//     'contractName.neofs' name
+//
+// TestRemoteStorageMigration stages:
+//  0. connect to remote blockchain by specified endpoint and run test local
+//     blockchain environment
+//  1. compile and deploy NeoFS NNS contract from sources located in '../nns' directory
+//  2. fetch tested contract executable by contractName parameter
+//  3. deploy tested contract in the test blockchain
+//  4. compile tested contract from sources located in '../contractName' directory
+//  5. update the tested contract with executables compiled in step 4
+//  6. check that test contract's storage is empty
+//
+// See Options to tune the default execution.
+func TestRemoteStorageMigration(tb testing.TB, blockChainRPCEndpoint, contractName string, opts Options) {
```

Comment:
As already discussed the "remoteness" of this procedure is better be handled by a script downloading data from this external source. Actual tests should operate with static data set stored somewhere in `testdata`.